### PR TITLE
Document defaults for gradle properties in Build Enviroment page in the user guide

### PR DESCRIPTION
**Pull request by tresat**
Addresses https://github.com/gradle/gradle/issues/15203 by adding defaults directly to properties page.

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [x] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

File: subprojects/docs/src/docs/userguide/running-builds/build_environment.adoc [hunk 1]
```diff
@@ -47,43 +47,55 @@ The following properties can be used to configure the Gradle build environment:
...
 When set to true, Gradle will reuse task outputs from any previous build, when possible, resulting is much faster builds. Learn more about <<build_cache.adoc#build_cache, using the build cache>>.
+_By default, the build cache is *not* enabled._
 
 `org.gradle.caching.debug=(true,false)`::
 When set to true, individual input property hashes and the build cache key for each task are logged on the console. Learn more about <<build_cache.adoc#sec:task_output_caching, task output caching>>.
+_Default is `false`._
 
 `org.gradle.configureondemand=(true,false)`::
 Enables incubating <<multi_project_configuration_and_execution.adoc#sec:configuration_on_demand, configuration on demand>>, where Gradle will attempt to configure only necessary projects.
+_Default is false._
```
Comment (big-guy) [inline subprojects/docs/src/docs/userguide/running-builds/build_environment.adoc:58]:
```suggestion
_Default is `false`._
```

File: subprojects/docs/src/docs/userguide/running-builds/performance.adoc [hunk 2]
```diff
@@ -186,7 +186,7 @@ project.tasks.register('copyFiles', Copy) { Task t ->
 ----
 
 You can see how this example uses the `register()` and `getByName()` methods, which are available on all Gradle “domain object containers”, like tasks, configurations, dependencies, extensions, etc.
-Some collections have dedicated types, `TaskContainer` being one of them, that have useful extra methods like the `create()` method above that takes a task type.
+Some collections have dedicated types, `TaskContainer` being one of them, that have useful extra methods like the link:{groovyDslPath}/org.gradle.api.tasks.TaskContainer.html#https://docs.gradle.org/current/dsl/org.gradle.api.tasks.TaskContainer.html#org.gradle.api.tasks.TaskContainer:create(java.lang.String,%20java.lang.Class)[create] method that takes a task type.
```
Comment (big-guy) [inline subprojects/docs/src/docs/userguide/running-builds/performance.adoc:189]: I think you might mean this? I didn't try it:
```suggestion
Some collections have dedicated types, `TaskContainer` being one of them, that have useful extra methods like the link:{groovyDslPath}/org.gradle.api.tasks.TaskContainer.html#org.gradle.api.tasks.TaskContainer:create(java.lang.String,%20java.lang.Class)[create] method that takes a task type.
```

**Review by big-guy — CHANGES_REQUESTED** [review]

**Comment by tresat** [comment]
@bot-gradle test this

File: subprojects/docs/src/docs/userguide/running-builds/performance.adoc [hunk 3]
```diff
@@ -186,7 +186,7 @@ project.tasks.register('copyFiles', Copy) { Task t ->
 ----
 
 You can see how this example uses the `register()` and `getByName()` methods, which are available on all Gradle “domain object containers”, like tasks, configurations, dependencies, extensions, etc.
-Some collections have dedicated types, `TaskContainer` being one of them, that have useful extra methods like the `create()` method above that takes a task type.
+Some collections have dedicated types, `TaskContainer` being one of them, that have useful extra methods like the link:{groovyDslPath}/org.gradle.api.tasks.TaskContainer.html#org.gradle.api.tasks.TaskContainer.html#org.gradle.api.tasks.TaskContainer:create(java.lang.String,%20java.lang.Class)[create] method that takes a task type.
```
Comment (big-guy) [inline subprojects/docs/src/docs/userguide/running-builds/performance.adoc:189]:
```suggestion
Some collections have dedicated types, `TaskContainer` being one of them, that have useful extra methods like the link:{groovyDslPath}/org.gradle.api.tasks.TaskContainer.html#org.gradle.api.tasks.TaskContainer:create(java.lang.String,%20java.lang.Class)[create] method that takes a task type.
```

**Comment by big-guy** [comment]
@bot-gradle test and merge

**Review by big-guy — APPROVED** [review]

**Pull request merged**